### ISSUE TITLE
Crux usage is broken

### ISSUE THREAD

**Issue by CharlesEGrant** [crux-toolkit/crux-toolkit#536]
The output of the crux usage message is broken. Any invocation of the usage message, 'crux help' or even just 'crux' with no arguments, or an invalid set of arguments results in the usage statement being printed, but then an endless stream of whitespace begins.

The code that prints the usage statement has been changed in ages, so I'm guessing that one of the recent additions to the usage statement is missing a key newline.

**Comment by wsnoble** [comment]
I ran the lint tests on Params.cpp, but it didn't turn up much:

```
../../src/util/Params.cpp:847:  Missing space after ,  [whitespace/comma] [3]
../../src/util/Params.cpp:860:  Missing space after ,  [whitespace/comma] [3]
../../src/util/Params.cpp:1303:  Missing space after ,  [whitespace/comma] [3]
../../src/util/Params.cpp:1306:  Missing space after ,  [whitespace/comma] [3]
../../src/util/Params.cpp:1767:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
../../src/util/Params.cpp:1768:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
../../src/util/Params.cpp:1769:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
../../src/util/Params.cpp:1770:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
../../src/util/Params.cpp:1772:  Weird number of spaces at line-start.  Are you using a 2-space indent?  [whitespace/indent] [3]
Done processing ../../src/util/Params.cpp
Total errors found: 9
```

**Comment by bmx8177** [comment]
Looks like the most recent change in the param-medic description broke it (based off this screen shot). You can see the new line starts there. It looks like Bill changed the ParamMedicApplication::getDescription() line.

I think we are missing a short description that is put in [[nohtml

![Capture](https://user-images.githubusercontent.com/22140743/144964035-6e9ceadf-07b4-4ac2-9bde-b4364ff2eb01.PNG)

**Comment by wsnoble** [comment]
Doh! I am totally busted!

Andy, can you please review this patch?  It also fixes some missing options in param-medic (issue #504).  Thanks!
[parammedic.txt](https://github.com/crux-toolkit/crux-toolkit/files/7665862/parammedic.txt)

**Comment by bmx8177** [comment]
I took a look and I have one small edit. Please remove '[[html:<p>]]' in the line below (line 210 in src/app/ParamMedicApplication.cpp

"[[html:<p>]]Examine the spectra in a file to estimate the best precursor"

Smoke tests all pass. This issue and issue #536 can be closed when it has been pushed.

**Comment by wsnoble** [comment]
Oops -- the above didn't show up. Let me try again.

I think we need the `<p>` in the HTML version to match the `</p>` in line 227.  Is there a reason you didn't want this paragraph enclosed in `<p>...</p>`?

**Comment by bmx8177** [comment]
oh, I didn't think about that. I was mostly confused about why we have '[[html:' two times. Once in line 210 and once in line 214. It seemed odd to me that we had two of the same tags in the same paragraph. 

I also just took a look at other the other applications do it and it is slightly different than how we are doing it here. It looks like there are two different descriptions. One tagged as [[html: and the other tagged as [[nohtml:. See example from tide-search below. We should probably follow that structure. 

![image](https://user-images.githubusercontent.com/22140743/145255425-a6cc3e2a-bdc1-4bac-87d8-33a0e2cd9993.png)

**Comment by wsnoble** [comment]
I was just trying to avoid having the same sentence repeated twice.  But do you like it better this way?

```
string ParamMedicApplication::getDescription() const {
  return
    "[[nohtml:Examine the spectra in a file to estimate the best precursor "
    "and fragment error tolerances for database search and to infer "
    "the presence of various types of modifications, including stable-isotope "
    "labeling, isobaric labeling, tandem mass tags, and enrichment of "
    "phosphorylated peptides.]]"
    "[[html:<p>Examine the spectra in a file to estimate the best precursor "
    "and fragment error tolerances for database search and to infer "
    "the presence of various types of modifications, including stable-isotope "
    "labeling, isobaric labeling, tandem mass tags, and enrichment of "
    "phosphorylated peptides.  For details on how param-medic works, "
    "see these publications:</p><blockquote>Damon H. May, Kaipo Tamura, "
    "William Stafford Noble. <a href=\"https://pubs.acs.org/doi/abs/10.1021/"
    "acs.jproteome.7b00028\">&quot;Param-Medic: A tool for improving MS/MS "
    "database search yield by optimizing parameter settings.&quot;</a> "
    "<em>Journal of Proteome Research</em>. 16(4):1817-1824, 2017.<br><br>"
    "Damon May, Kaipo Tamura, William Stafford Noble. <a href=\""
    "https://pubs.acs.org/doi/full/10.1021/acs.jproteome.8b00954\">"
    "&quot;Detecting modifications in proteomics experiments with "
    "Param-Medic.&quot;</a> <em>Journal of Proteome Research</em>. "
    "18(4):1902-1906, 2019.</blockquote>Note that you can access some of the "
    "functionality of Param-Medic by using the "
    "<code>--auto-precursor-window</code> and <code>--auto-mz-bin-width</code>"
    " options in <code>tide-search</code>.</p>]]";
}
```
I think my version is equivalent to the above.

**Comment by bmx8177** [comment]
Yes your version is equivalent to the above (I checked). I personally like this version better but I don't have a strong preference. More importantly is that we fixed the bug.
